package.py

```diff
@@ -0,0 +1,911 @@
+            encoding='utf8',
+            shell=True,
+        ).strip()
+
+        if output != 'master':
+            logger.info(f'Current branch is {output}, skipping check')
+            return
+
+        unmerged_commits = subprocess.check_output(
+            'git rev-list HEAD..bugfixes --no-merges | wc -l | xargs echo -n',
+            encoding='utf8',
+            shell=True,
+        ).strip()
+
+        if unmerged_commits != "0":
+            logger.error(
+                f"Found {unmerged_commits} in bugfixes that have not been merged for release",
+            )
+            exit(1)
+        logger.info("branch is up to date with bugfixes")
+
+    def check_environment(self) -> None:
+        if self.arch not in self.archs:
+            logger.error(f'{self.arch} is not supported, packaging only supports {self.archs}')
+            exit(1)
+
+        if self.os not in self.oses:
+            logger.error(
+                f'{platform.system()} is not supported, packaging only supports {self.oses}',
+            )
+            exit(1)
+
+        if not self.is_ci() and not os.environ.get('VIRTUAL_ENV'):
+            logger.error('The script should not run outside a virtual environment if not on CI')
+            exit(1)
+
+        self.check_repo()
+
+    def is_mac(self) -> bool:
+        return self.os == 'Darwin'
+
+    def is_linux(self) -> bool:
+        return self.os == 'Linux'
+
+    def is_windows(self) -> bool:
+        return self.os == 'Windows'
+
+    def is_mac_runner(self) -> bool:
+        return self.is_mac() and self.is_ci()
+
+    def is_universal2(self) -> bool:
+        return self.is_mac_runner() or os.environ.get('MACOS_BUILD_ARCH') == 'universal2'
+
+    def is_x86_64(self) -> bool:
+        return self.arch in ['x86_64', 'AMD64']
+
+    def os_tag(self) -> str:
+        if self.is_mac():
+            return 'macos'
+        if self.is_linux():
+            return 'linux'
+        if self.is_windows():
+            return 'windows.exe'
+        raise Exception(f'Invalid os {self.os}')
+
+    @staticmethod
+    def mac_electron_env_set() -> bool:
+        return os.environ.get('MACOS_ELECTRON_ARCH') is not None
+
+    def get_frontend_env(self) -> dict[str, str]:
+        if os.environ.get('GH_TOKEN'):
+            logger.info('GH_TOKEN WAS SET')
+        else:
+            logger.info('NO GH_TOKEN SET')
+        env = os.environ.copy()
+        if self.is_mac() and not self.mac_electron_env_set() and not self.is_ci():
+            if self.is_x86_64():
+                arch = 'x64'
+            else:
+                arch = 'arm64'
+
+            env.setdefault('MACOS_ELECTRON_ARCH', arch)
+        return env
+
+    @staticmethod
+    def is_ci() -> bool:
+        return os.environ.get('CI') is not None
+
+    @staticmethod
+    def update_path(directory: Path) -> None:
+        path = os.environ.get('PATH')
+        os.environ.setdefault('PATH', f'{path};{directory}')
```

Comment:
This method is not used or at least I don't see it in a search. Can be removed then